packages/pytest-simcore/src/pytest_simcore/postgres_service.py

```diff
@@ -58,27 +58,41 @@ def postgres_engine(
 
 @pytest.fixture(scope="module")
 def postgres_db(
-    postgres_dsn: Dict, postgres_engine: sa.engine.Engine
+    postgres_dsn: Dict,
+    postgres_engine: sa.engine.Engine,
 ) -> sa.engine.Engine:
-    # migrate the database
+
+    # upgrades database from zero
     kwargs = postgres_dsn.copy()
     pg_cli.discover.callback(**kwargs)
     pg_cli.upgrade.callback("head")
+
     yield postgres_engine
 
+    # downgrades database to zero ---
+    #
+    # NOTE: This step CANNOT be avoided since it would leave the db in an invalid state
+    # E.g. 'alembic_version' table is not deleted and keeps head version or routines
+    # like 'notify_comp_tasks_changed' remain undeleted
+    #
     pg_cli.downgrade.callback("base")
-    pg_cli.clean.callback()
+    pg_cli.clean.callback()  # just cleans discover cache
 
-    # FIXME: deletes all because downgrade is not reliable!
+    # FIXME: migration downgrade fails to remove User types SEE https://github.com/ITISFoundation/osparc-simcore/issues/1776
+    # Added drop_all as tmp fix
     metadata.drop_all(postgres_engine)
 
 
 @pytest.fixture(scope="module")
 def postgres_session(postgres_db: sa.engine.Engine) -> sa.orm.session.Session:
-    Session = sessionmaker(postgres_db)
-    session = Session()
+    from sqlalchemy.orm.session import Session

```

Comment:
is there a reason why this is imported here, or why `sa.orm.session.Session` was not used as a type?

Reply:
We agreed not to use sqlalchemy ORMs (aiopg uses a lower level).
 This is an old fixture that we should replace and I put the import inside to keep it all together until we remove it...